modules/extension/wmts/src/main/java/org/geotools/ows/wmts/client/WMTSTile.java

```diff
@@ -180,16 +179,28 @@ private URL getKVPurl(String baseUrl, TileIdentifier tileIdentifier) throws Runt
         params.put("TileRow", tileIdentifier.getY());
 
         StringBuilder arguments = new StringBuilder();
+        // in theory it should end with ? but there may be other params that the
+        // server needs out of spec
+        boolean skipFirst = false;
+        if (!baseUrl.contains("?")) {
+            arguments.append("?");
+            skipFirst = true;
```

Comment:
You could simplify by using this logic to determine the separator to use next:
```
String seperator = "&";
if (!baseUrl.contains("?")) {
   String seperator = "?";
}
```

The code should be easier to follow:
```
arguments
   .append( separator )
   .append( key )
   .append( = )
   .append(URLEncoder.encode(val.toString(), "UTF-8"));
separator = '&'
```

This also prevents `?` being appended in the event no useful params are provided.

Reply:
@jodygarnett Thanks for the feedback. I've made some new commits.